sklearn/metrics/_plot/tests/test_confusion_matrix_display.py

```diff
@@ -267,6 +267,23 @@ def test_confusion_matrix_contrast(pyplot):
     assert_allclose(disp.text_[1, 1].get_color(), min_color)
 
 
+def test_confusion_matrix_with_nan_values_contrast(pyplot):
+    """Text values are displayed in contrasting colors for matrices with nans.
+
+    This is a regression test for #27306
+    """
+    cm = np.array([[1, 2], [3, np.nan]])
+    disp = ConfusionMatrixDisplay(cm)
+
+    disp.plot(cmap=pyplot.cm.Blues)
+    min_color = pyplot.cm.Blues(0)
+    max_color = pyplot.cm.Blues(255)
+    assert_allclose(disp.text_[0, 0].get_color(), max_color)
+    assert_allclose(disp.text_[0, 1].get_color(), min_color)
+    assert_allclose(disp.text_[1, 0].get_color(), min_color)
```

Comment:
For my education: why don't the values 1, 2 and 3 have different colours? I guess `[0, 0]` corresponds to `1` in `cm`, so that gets `max_color` (should we change the name to `min_color` as 1 is the smallest value?) But then 2 and 3 are both `min_color`.

Overall, I'm puzzled :D

Reply:
Haha, I'll try to explain. So the values can only have two possible colors, `cmap_min` or `cmap_max`. All values that are below the threshold value get the `cmap_max` color, and all values equal or above the threshold value get the `cmap_min` color.

I kept the names `min_color` and `max_color` in line with this. Because the value `1` is below the threshold of `2`, the value should get the `cmap_max` value.

Does that make sense?